src/audio/selector.c

```diff
@@ -93,9 +93,9 @@ static int sel_set_channel_values(struct comp_data *cd, uint32_t in_channels,
 		return -EINVAL;
 	}
 
-	cd->in_channels_count = in_channels;
-	cd->out_channels_count = out_channels;
-	cd->sel_channel = ch_idx;
+	cd->config.in_channels_count = in_channels;
```

Comment:
This should really follow existing naming convention where we use source and sink instead of in and out respectively. This could be a separate patch though..

Reply:
OK, I'll make it as a separate patch